include/nighthawk/common/request_source_plugin.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "envoy/api/api.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+
+#include "nighthawk/common/request_source.h"
+
+namespace Nighthawk {
+
+/**
+ * An interface for different RequestSource plugins.
+ * Uses a plugin-specific config proto.
+ */
+class RequestSourcePlugin : RequestSource {
+public:
+  RequestSourcePlugin(Envoy::Api::Api& api) : api_(api) {}
```

Comment:
Mocking might be easier if the api argument is added via a setter or separate init call. It may even void the need to be stateful with respect to keeping a reference to the api.

Reply:
Also, aren't the things we needed to construct an implementation specific to that implementation?
Maybe consider passing a RequestSourcePluginContext(Ptr) to abstract that away here? 
Or if we don't want to impose an interface for that, just a (void*)? Otherwise, we can expect to have to update the interface as implementations are added.
Then at the implementation layer plugins could use (dynamic_)casts to get to a specific re-implementation of it to satisfy their needs?